velox/exec/ExchangeQueue.h

```diff
@@ -26,7 +26,8 @@ class SerializedPage {
   // Construct from IOBuf chain.
   explicit SerializedPage(
       std::unique_ptr<folly::IOBuf> iobuf,
-      std::function<void(folly::IOBuf&)> onDestructionCb = nullptr);
+      std::function<void(folly::IOBuf&)> onDestructionCb = nullptr,
+      int64_t rows = -1);
```

Comment:
nit: If the default value of `rows` is `0`, then we no longer need a lot of `VELOX_CHECK(data.rows() >= 0)` when getting its row count, right?

Reply:
Nop. The `row` param is not always provided, because for `ExchangeSource` the rows of a serializedPage is not known until the page gets unserialized. So the `row` param should be optional, and `-1` is necessary to indicates if the `row` param is valid or not.